### ISSUE TITLE
Support CPUBurst option in systemd.resource-control

### ISSUE THREAD

**Issue by Zheaoli** [systemd/systemd#26658]
### Component

other

### Is your feature request related to a problem? Please describe

The CPU Burst Option has been supported in Linux Kernel Cgroup since 5.15. FYI  https://www.kernel.org/doc/html/latest/scheduler/sched-bwc.html#burst-feature

But the systemd does not support this feature for service resource control.

### Describe the solution you'd like

Allow people to set the `CPUBurst` option for service

**Comment by Werkov** [comment]
`CPUQuota=` is in percents and uses implicit period of 100 ms.
I wonder how is burst value determined from the user perspective (absolute or relative). Do you have any advice regarding that?

How should the burst value passed to systemd? (Percents relative to the same period as quota?)

**Comment by Zheaoli** [comment]
> `CPUQuota=` is in percents and uses implicit period of 100 ms. I wonder how is burst value determined from the user perspective (absolute or relative). Do you have any advice regarding that?
> 
> How should the burst value passed to systemd? (Percents relative to the same period as quota?)

Yep, I think "use Percents relative to the same period as quota" is a good idea

**Comment by Werkov** [comment]
PR would be welcomed :-)

**Comment by Zheaoli** [comment]
OK, Please assign this issue to me

**Comment by kolyshkin** [comment]
@Zheaoli were you able to work on that?

**Comment by d-uzlov** [comment]
I, as a user, strongly believe that CPUBurst should be set as an absolute value in [milli]seconds.
Burst is most useful in cases when CPUQuota is low because CPU usage on average is low, but the application sleeps most of the time, and works in short bursts.
If the application needs 200 ms burst to do its work, it doesn't matter if period is 10 ms or 100 ms.
As I see it, CPU burst as a concept was created to decouple burst duration from quota and period. It would be strange to link it back to period via percentage value.

However, I should note that CPU burst option is currently broken (or half-baked) in Linux, and neither `burst=duration_ms` nor `burst=period%` will work on modern mainstream Linux kernels.
The burst value is artificially [limited to CPUQuota as upper boundary](https://github.com/torvalds/linux/blob/1b237f190eb3d36f52dffe07a40b5eb210280e00/kernel/sched/core.c#L9893). If quota is 10000, then setting burst above 10000 will result in an error.

So, for systems running the mainstream kernel, the only practical way to implement this would be to do `burst=quota%`, with upper limit of 100%. Which would be strange and kinda useless, if you ask me.
Or `burst=period%` with upper limit of `CPUQuota`, which is also really strange, and gives me a question of "Why not just increase CPUQuota?"
